transaction-gossip.md

```diff
@@ -65,14 +54,30 @@ proof       := TODO
 transaction := TODO
 ```
 
+## Secondary Routing Table
+
+Clients in the Transaction Gossip Network are expected to maintain both the standard routing table and a secondary routing table that is anchored to node radius values.
+
+The secondary routing table is subject to all of the same maintenance and management rules as the primary table.  Any node that is added to the secondary routing table must also satisfy the following validity condition:
+
+```
```

Comment:
My concern with this approach was when trying to think about malicious nodes.  A malicious node could monitor inbound `Ping` messages and always respond with `Pong.radius = Ping.radius + random()` to ensure that it is *always* a routing table candidtate.  This would allow a malicious node to over-represent itself in other node's routing tables.

This approach seems to increase the ability for malicious nodes to do things like disrupting tx propagation.

I gave a similar idea towards adding a preference towards `dist(node.node_id, content_id) == node.transaction_radius` which aims to populate the table with nodes that are specifically "far away" which results in the furthest bucket being mostly full radius nodes.  I think this suffers from the same problem of over-representation and incentives for malicious nodes to lie.

Reply:
Sure. I am not firm on what is the best approach for the secondary routing table. I brought that up primarily to provide alternatives. I would also prefer to see the construction of the routing table to be similar for networks other than "transaction" gossip. 

With regard to "node lying to over-represent", two thoughts. One is about "eviction" policy. DHT routing table usually has an eviction policy to control peers that are not doing much or acting strange. A node that represents its radius will fail to answer queries that it is supposed to answer. Second is that there is a cost to be associated with being a peer with more and more nodes; it requires resources to maintain those connections. For example, a node could always creates more fake identities, and participate in the network as many versions of itself. That is a form of over-representation. Nodes do not do that because the gain is probably not higher than the costs. I am not all that convinced that there is much to be gained from cheating on this network.